code/nfts/create-nft-token/create-nft-token.en.ts

```diff
@@ -0,0 +1,59 @@
+import {
+    Connection,
+    Keypair,
+    clusterApiUrl,
+    LAMPORTS_PER_SOL,
+} from "@solana/web3.js";
+
+import { Token, TOKEN_PROGRAM_ID } from "@solana/spl-token";
+
+(async () => {
+    const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
+
+    const userKeypair = Keypair.generate();
+
+    // Airdrop SOL for the user to create the NFT
+    const airdropSignature = await connection.requestAirdrop(
+        userKeypair.publicKey,
+        5 * LAMPORTS_PER_SOL
+    );
+    await connection.confirmTransaction(airdropSignature, "confirmed");
+
+    // Create the Mint Account for the NFT
+    const mintAccount = await Token.createMint(
+        connection,
+        userKeypair,
+        userKeypair.publicKey,
+        null,
+        0,
+        TOKEN_PROGRAM_ID
+    );
+
+    // Get/Create the Associated Account for the user to hold the NFT
+    const userAssosciatedAccount =
+        await mintAccount.getOrCreateAssociatedAccountInfo(
+            userKeypair.publicKey
+        );
+
+    // Mint 1 token to the user's associated account
+    await mintAccount.mintTo(
+        userAssosciatedAccount.address,
+        userKeypair.publicKey,
+        [],
+        1
+    );
+
+    // Reset mint_authority to null from the user to prevent further minting
+    await mintAccount.setAuthority(
+        mintAccount.publicKey,
+        null,
+        "MintTokens",
+        userKeypair.publicKey,
+        []
+    );
+
+    // Checking balance of the user's associated account
+    const balance = await mintAccount.getAccountInfo(
```

Comment:
So I ran this and it failed due to `amount` not available on the promise. You can solve by either wrapping the getAccountInfo call or doing something like this:

```
const mintAccountInfo = await mintAccount.getAccountInfo(
      userAssosciatedAccount.address
  );
  console.log(mintAccountInfo.amount.toNumber());
1
  ```

Reply:
Hey, sorry for the delay. I was out for the weekend and couldn't get much work done. I fixed the issue mentioned above and I'll get the rest of the tasks done ASAP.